src/nunavut/lang/c/support/serialization.j2

```diff
@@ -33,29 +33,45 @@
 
 #ifndef NUNAVUT_SUPPORT_SERIALIZATION_H_INCLUDED
 #define NUNAVUT_SUPPORT_SERIALIZATION_H_INCLUDED
-
+{% if options.cpp_safe_c %}
 #ifdef __cplusplus
 #   if (__cplusplus < 201402L)
 #       error "Unsupported language: ISO C11, C++14, or a newer version of either is required."
 #   endif
 extern "C"
 {
-#else
+#else{% endif %}
 #   if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
 #       error "Unsupported language: ISO C11 or a newer version is required."
 #   endif
+{%- if options.cpp_safe_c %}
 #endif
 
+#ifdef __cplusplus
+#include <cstring>
+{% if not options.omit_float_serialization_support %}
+#include <cfloat>
+#include <cmath>  // For isfinite().
+using std::isfinite;
+{% endif -%}
+#include <cstdint>
+#include <cassert>  // For _Static_assert (C11) static_assert (C23) and assert() if NUNAVUT_ASSERT is used.
```

Comment:
What is the rationale behind replacing c-style headers with c++-style ones? They have been undeprecated in C++23 so both should be valid. Unless you have a special toolchain that comes with a standard library without the .h files...

Reply:
(thinking...)